torch/distributions/constraints.py

```diff
@@ -275,6 +276,17 @@ def check(self, value):
         return lower_triangular & positive_diagonal
 
 
+class _CorrCholesky(Constraint):
+    """
+    Constrain to lower-triangular square matrices with positive diagonals and each
+    row vector being of unit length.
+    """
+    def check(self, value):
+        row_norm = torch.norm(value, dim=-1)
+        unit_row_norm = (row_norm <= 1. & row_norm >= 1e-6).all(dim=-1)
```

Comment:
I guess you meant `1 - 1e-6`.

nit: `torch.norm` is deprecated https://pytorch.org/docs/stable/generated/torch.norm.html